services/storehost/storagemonitor.go

```diff
@@ -71,35 +87,39 @@ type (
 	}
 )
 
-// Monitoring housekeeping will happen every 2 minutes
-const storageMonitoringInterval = time.Duration(2 * time.Minute)
-
-const bytesPerMB = 1024 * 1024
-
-const warningThreshold = 0.1        // 10%
-const alertThreshold = 0.05         // 5%
-const resumeWritableThreshold = 0.2 // 20%
-
 // NewStorageMonitor returns an instance of NewStorageMonitor.
 func NewStorageMonitor(store *StoreHost, m3Client metrics.Client, hostMetrics *load.HostMetrics, logger bark.Logger, path string) StorageMonitor {
 	return &storageMonitor{
-		storeHost:        store,
-		logger:           logger,
-		m3Client:         m3Client,
-		hostMetrics:      hostMetrics,
-		monitoringTicker: time.NewTicker(storageMonitoringInterval),
-		monitoringPath:   path,
-		mode:             SMReadWrite,
+		storeHost:      store,
+		logger:         logger,
+		m3Client:       m3Client,
+		hostMetrics:    hostMetrics,
+		monitoringPath: path,
+		mode:           SMReadWrite,
 	}
 }
 
 // Start starts the monitoring
 func (s *storageMonitor) Start() {
-	s.logger.Info("StorageMonitor: started")
 
 	s.closeChannel = make(chan struct{})
 
-	go s.doHouseKeeping()
+	go func() {
+
+		ticker := time.NewTicker(storageMonitoringInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				go s.checkStorage()
```

Comment:
Why spin another goroutine?

Reply:
Well, that's how it used to be ..

In my slightly 'cleaned up' version in PR #322 (https://github.com/uber/cherami-server/pull/322/files#diff-cb966d6a28a81db47b41fccd069cfe0b), I don't spin up a new go-routine.